lib/SIL/SILVerifier.cpp

```diff
@@ -577,6 +578,38 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     SILLocation::LocationKind LocKind = L.getKind();
     ValueKind InstKind = I->getKind();
 
+    // Check that there is at most one debug variable defined
+    // for each argument slot.
+    if (!DS->InlinedCallSite) {
+      SILDebugVariable VarInfo;
+      if (auto *DI = dyn_cast<AllocStackInst>(I)) {
+        VarInfo = DI->getVarInfo();
+      } else if (auto *DI = dyn_cast<AllocBoxInst>(I)) {
+        VarInfo = DI->getVarInfo();
+      } else if (auto *DI = dyn_cast<DebugValueInst>(I)) {
+        VarInfo = DI->getVarInfo();
+      } else if (auto *DI = dyn_cast<DebugValueAddrInst>(I)) {
+        VarInfo = DI->getVarInfo();
+      }
+
+      auto ArgNo = VarInfo.ArgNo;
+      if (ArgNo) {
+        // It is a function argument.
+        if (ArgNo < DebugVars.size() && DebugVars[ArgNo].ArgNo) {
+          require(DebugVars[ArgNo].Name == VarInfo.Name,
+                  "Re-definition of a debug variable");
```

Comment:
Unfortunately, we it does not support format strings. It was my first idea too ;-)